src/components/DataDisplayer/index.jsx

```diff
@@ -7,14 +7,20 @@ import {
   MagicCardBox,
 } from "./style";
 import MagicCard from "../MagicCard";
-import useFetch from "../../utils/fetcher";
+import { useSelector, useDispatch } from "react-redux";
+import { fetchData } from "../../utils/fetcher";
 import usePagination from "../../utils/paginator";
 
 export default function DataDisplayer({ url, name, Content }) {
-  const { data, error, loading } = useFetch(url);
+  const dispatch = useDispatch();
+  const { data, loading, error } = useSelector((state) => state.fetch);
   const { totalPages, paginatedData, currentPage, handlePageChange } =
     usePagination(data);
 
+  useEffect(() => {
+    dispatch(fetchData(url));
```

Comment:
better to _if_ check for url 
just in case url isn't there for some reason, you don't want to call dispatch in this situation